Athena/Athena/Windows/ReturnBookWindow.xaml.cs

```diff
@@ -51,16 +51,22 @@ public string ToAuthorsNames(Book book)
             return builder.ToString();
         }
 
-        private void ReturnBorrowedBook_Click(object sender, RoutedEventArgs e)
+        private async void ReturnBorrowedBook_Click(object sender, RoutedEventArgs e)
         {
-            Borrowing.ReturnDate = Calendar.SelectedDate.Value;
-            ApplicationDbContext.Instance.Entry(Borrowing).State = EntityState.Modified;
-            ApplicationDbContext.Instance.SaveChanges();
-            BookReturned?.Invoke(this, EventArgs.Empty);
+            var borrowing = await ApplicationDbContext.Instance.Borrowings.FindAsync(Borrowing.Id);
+            borrowing.ReturnDate = Calendar.SelectedDate;
+            await ApplicationDbContext.Instance.SaveChangesAsync();
+            BookReturned?.Invoke(this, new EventArgs<Borrowing>(borrowing));
             this.Close();
         }
-
-        public event EventHandler BookReturned;
-
+        public class EventArgs<Borrowing> : EventArgs
```

Comment:
Skąd taka nazwa? Wygląda fajnie. Jeśli gdzieś na necie polecali to spoko, jak nie to zmień nazwę  na `BookReturnedEventArgs`